tests/unit/options.js

```diff
@@ -1745,6 +1736,42 @@ exports.strict = function (test) {
   test.done();
 };
 
+/**
+ * This test asserts sub-optimal behavior.
+ *
+ * In the "browserify", "node" and "phantomjs" environments, user code is not
+ * executed in the global scope directly. This means that top-level `use
+ * strict` directives, although seemingly global, do *not* enable ES5 strict
+ * mode for other scripts executed in the same environment. Because of this,
+ * the `strict` option should enforce a top-level `use strict` directive in
+ * those environments.
+ *
+ * The `strict` option was implemented without consideration for these
+ * environments, so the sub-optimal behavior must be preserved for backwards
+ * compatability.
+ *
+ * TODO: Interpret `strict: true` as `strict: global` in the Browserify,
+ * Node.js, and PhantomJS environments, and remove this test in JSHint 3
+ */
+exports.strictEnvs = function (test) {
+  var partialStrict = [
+    "void 0;",
+    "(function() { void 0; }());",
+    "(function() { 'use strict'; void 0; }());"
+  ];
+  TestRun(test, "")
+    .addError(2, "Missing \"use strict\" statement.")
+    .test(partialStrict, { strict: true, browserify: true });
+  TestRun(test, "")
+    .addError(2, "Missing \"use strict\" statement.")
+    .test(partialStrict, { strict: true, node: true });
+  TestRun(test, "")
+    .addError(2, "Missing \"use strict\" statement.")
```

Comment:
I agree with the sentiment, but this is unavoidable because [the assertion uses a strict equality to compare "expected" and "actual" string values](https://github.com/jshint/jshint/blob/cfe6c43548debf11b4f7b4098d79a1316ca420a7/tests/helpers/testhelper.js#L67)